src/app/period.ts

```diff
@@ -119,6 +119,10 @@ export class ImmutablePeriod implements ImmutablePeriodIF {
     return new ImmutablePeriod({...this, lastUpdateUUID: lastUpdateUUID});
   }
 
+  withNewBuckets(newBuckets: readonly ImmutableBucket[]): ImmutablePeriod {
```

Comment:
I believe this function can be private - is that right?

I also think if we're going to abstract this code, there are a number of other places in the file we can use it.